internal/provutils/slices.go

```diff
@@ -0,0 +1,29 @@
+package provutils
+
+// FindMissing returns all elements of the required list that are not found in the entries list.
+// Duplicate entries in required do not require duplicate entries in toCheck.
+// E.g. FindMissing([a, b, a], [a]) => [b], and FindMissing([a, b, a], [b]) => [a, a].
+//
+// See also: FindMissingFunc.
+func FindMissing[S ~[]E, E comparable](required, toCheck S) S {
```

Comment:
Is there something that prevents us from using the standard set? https://pkg.go.dev/github.com/golang-collections/collections/set#Set.Difference

Reply:
`Set`s use a map for the backend which makes them nondeterministic.

Reply:
I was thinking of python rather than golang. It's not in the standard library, and the link I provided is not the standard collections library.